### PR TITLE
New CI

### PR DESCRIPTION
**Pull request by stefanotorresi**
Sister PR of ClusterLabs/ha_cluster_exporter#163, most of the code was copy-pasted from there.

### CODE REVIEW

**Comment by stefanotorresi** [comment]
this is now ready for review, but please defer merging until the secrets are properly set in GH.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,14 +1,69 @@
...
+## OBS Packaging
 
-The CI will automatically interact with SUSE's [Open Build Service](https://build.opensuse.org): the master branch will be kept in sync with the `server:monitoring` project (the development upstream), while a new Submit Request against the `openSUSE:Factory` project (the stable downstream) can be triggered by publishing a new GitHub release or pushing a new Git tag.
- 
-When releasing to `openSUSE:Factory`, please ensure that tags always follow the [SemVer](https://semver.org/) scheme, and that [the changelog](prometheus-hanadb_exporter.changes) contains a new entry, otherwise the request submission might fail.
+The CI will automatically interact with SUSE's [Open Build Service](https://build.opensuse.org): the `master` branch will be kept in sync with the `network:ha-clustering:sap-deployments:devel` project.
+
+### Publishing the Exporter RPM in openSUSE code stream.
+
+For the exporter only, a new Submit Request against the `server:monitoring` project can be triggered by publishing a new GitHub release.
```
Comment (arbulu89) [inline CONTRIBUTING.md:9]: Is `server:monitoring` still used?

Comment (stefanotorresi) [reply CONTRIBUTING.md:9]: Yeah, the exporter packages released in `openSUSE:Factory` are still linked to `server:monitoring` as dev upstream.
I will propose to re-structure all our projects in OBS in the future.

File: .github/workflows/dashboards-ci.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - 'packaging/obs/grafana-sap-hana-dashboards/**'
+    - '.github/workflows/dashboards*'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v1
+    - run: npm -g install jsonlint
```
[the remark this answers is not in the document]
Comment (stefanotorresi) [reply .github/workflows/dashboards-ci.yml:27]: I don't know any easier way to get ahold of a JSON linter :)

**Comment by stefanotorresi** [comment]
> * The travis badge in the README should be replaced by a github one (if it exists)

Done! For the time being, I didn't replace the codeclimate one, though. I will explore this in the future for all the exporters.

> * Now it looks like the `changes` file is updated from the github release, but we still have the `changes` file in the source code. How is this working? Maybe we could describe this in the README. Personally, I'm not a big fan of this process (update changes fro GH), but let's see how it goes

This was actually a mistake on my part: I have removed the changes file, which is now managed automatically. This only applies to the exporter, though.

I have left the README mostly un-touched here, but I have another PR coming up that overhauls it entirely, because some instructions are outdated.

> * How is the python matrix working? I see that you define the python versions in the tox file and in the github workflow file.

The two are not strictly related. We run the `test` job in both python 3.6 and 3.8. We use the tox `py` env to automatically get the currently installed python version. The other tox envs are only useful to run tests locally in systems with multiple python versions installed.

> * I see that almost all the `osc` operations are done in the make file. The container for the CI was created exactly for that, to not duplicate the code in every project we have. If the commands go here it doesn't have sense to use the container anymore. I would use one or the other approach, but the combination of both looks strange.

The container is still needed to provide an environment with `osc` installed and configured. I haven't yet had the chance to move what is being done in the make targets into the scripts within the container, but this would certainly be a very welcomed improvement! For the time being, copy-pasting was the simplest approach.
The way forward would probably be to create a custom reusable GitHub action (which at the end of the day is just a container and a bash script with a specific interface).

> * Now the CI looks much more complicated than before, essentially because more files are involved. I guess that these files won't be changed at all, so it should be fine

Heh, that's the trade-off for using the same repo for two packages, but I think it's worth it. Making related changes to both the exporter and the dashboards is a much smoother process now, e.g. see ClusterLabs/ha_cluster_exporter#167.

File: .github/workflows/dashboards-ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (arbulu89) [reply .github/workflows/dashboards-ci.yml:27]: That's fine hehe
Just I was surprised to find it

**Pull request merged**